Improve pre-upgrade documentation (#3855)
theforeman/foreman-documentation: guides/common/modules/con_upgrade-path-overview.adoc

[id="upgrade-path-overview"]
= Upgrade path overview

Upgrading to {ProjectName} {ProjectVersion} from {ProjectName} {ProjectVersionPrevious} includes the following high-level steps:

ifdef::foreman-deb,foreman-el,orcharhino[]
. Upgrading your {ProjectServer} to {ProjectVersion}.
. Upgrading your {SmartProxyServers} to {ProjectVersion}.
endif::[]
ifdef::katello,satellite[]
. Ensuring that your {ProjectServer}s and {SmartProxyServers} are running on {Project} {ProjectVersionPrevious}.
. Upgrading your {ProjectServer}:
.. Upgrading your {ProjectServer} to {ProjectVersion}.
.. Synchronizing the new {ProjectVersion} repositories.
. Upgrading your {SmartProxyServers} to {ProjectVersion}.
endif::[]

ifndef::orcharhino[]

> You can upgrade your {SmartProxies} separately over multiple maintenance windows because {SmartProxies} at version {ProjectVersionPrevious} and {ProjectVersionPrevious-Previous} will keep working with your upgraded {ProjectServer} {ProjectVersion}.

Reviewer: ```suggestion
You can upgrade your {SmartProxies} separately over multiple maintenance windows because versions {ProjectVersionPrevious} and {ProjectVersionPrevious-Previous} remain compatible with your upgraded {ProjectServer} {ProjectVersion}.
```

endif::[]
ifdef::orcharhino[]

> You can upgrade your {SmartProxies} separately over multiple maintenance windows because {SmartProxies} at version {ProjectVersionPrevious} will keep working with your upgraded {ProjectServer} {ProjectVersion}.

Reviewer: ```suggestion
You can upgrade your {SmartProxies} separately over multiple maintenance windows because {SmartProxies} running version {ProjectVersionPrevious} remain compatible with {ProjectServer} {ProjectVersion}.
```

endif::[]
Upgrading {SmartProxies} separately can be useful in the following situations:

* If you want to have several smaller outage windows instead of one larger window.
* If {SmartProxies} in your organization are managed by several teams and are located in different locations.
ifdef::katello,orcharhino,satellite[]
* If you use a load-balanced configuration, you can upgrade one load-balanced {SmartProxy} and keep other load-balanced {SmartProxies} at one version lower.
This allows you to upgrade all {SmartProxies} one after another without any outage.
endif::[]

ifdef::orcharhino[]
{SmartProxies} at version {ProjectVersionPrevious} retain all of their previous functionality.
endif::[]
ifndef::orcharhino[]
{SmartProxies} at version {ProjectVersionPrevious} and {ProjectVersionPrevious-Previous} retain all of their previous functionality.
endif::[]

> New functionality added in the {ProjectVersion} release will not work until you upgrade {SmartProxies} to {ProjectVersion}.

Reviewer: ```suggestion
New functionality added in the {ProjectVersion} release is available only after you upgrade your {SmartProxies} to {ProjectVersion}.
```
